tracer/src/Datadog.Trace.Tools.Runner/CheckAgentCommand.cs

```diff
@@ -0,0 +1,54 @@
+// <copyright file="CheckAgentCommand.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+#nullable enable
+
+using System;
+using System.Threading.Tasks;
+using Datadog.Trace.Agent;
+using Datadog.Trace.Configuration;
+using Datadog.Trace.Tools.Runner.Checks;
+using Spectre.Console;
+using Spectre.Console.Cli;
+
+namespace Datadog.Trace.Tools.Runner
+{
+    internal class CheckAgentCommand : AsyncCommand<CheckAgentSettings>
+    {
+        public override async Task<int> ExecuteAsync(CommandContext context, CheckAgentSettings settings)
+        {
+            ExporterSettings configuration;
+
+            if (settings.Url == null)
+            {
+                // Try to autodetect the agent settings
+                configuration = new ExporterSettings(new EnvironmentConfigurationSource());
+
+                AnsiConsole.WriteLine($"No agent url provided, using environment variables");
+            }
+            else
+            {
+                configuration = new ExporterSettings { AgentUri = new Uri(settings.Url) };
+
+                // TODO: Remove when the logic has been moved to the ImmutableExporterSettings constructor
+                if (settings.Url.StartsWith("unix://"))
```

Comment:
Seems slightly odd to me to support UDS here but not windows pipes, but then we have separate settings for all those currently, so I guess that's just the way it is 🤷

Reply:
Yeah. We _could_ support them, but named pipes are only used in AAS, and I don't think it's practical to run the tool there. So given that it's not completely straightforward to add named pipes support, let's leave it as is until somebody actually needs it.

Reply:
>named pipes are only used in AAS

Named pipes are only used by _default_ in AAS right, they can be used outside? 

> let's leave it as is until somebody actually needs it

Agreed

Reply:
> Named pipes are only used by default in AAS right, they can be used outside?

They definitely _can_, but that's an undocumented feature so I think it's safe to assume nobody is using it outside of AAS